### PR TITLE
Support `login` from profile with `credential_process`

### LINKED ISSUE #1134: [Feature Request] Support `aws-vault login` for `credential_process` returning temporary credentials
Similar to #1087, I would like to use aws-vault to start console sessions from cached credentials created by an external process.

### PR DESCRIPTION
**Pull request by christophgysin**
Fixes #1134

### CODE REVIEW

File: vault/vault.go [hunk 1]
```diff
@@ -269,14 +269,8 @@ func NewTempCredentialsProvider(config *Config, keyring *CredentialKeyring) (aws
...
-func NewFederationTokenCredentialsProvider(ctx context.Context, profileName string, k *CredentialKeyring, config *Config) (aws.CredentialsProvider, error) {
-	credentialsName, err := FindMasterCredentialsNameFor(profileName, k, config)
-	if err != nil {
-		return nil, err
-	}
-	masterCreds := NewMasterCredentialsProvider(k, credentialsName)
-
-	return NewFederationTokenProvider(ctx, masterCreds, config)
+func NewFederationTokenCredentialsProvider(ctx context.Context, profileName string, keyring *CredentialKeyring, config *Config) (aws.CredentialsProvider, error) {
+	return NewTempCredentialsProvider(config, keyring)
```
Comment (mtibben) [inline vault/vault.go:273]: A bit confused by this - what's the point of the `NewFederationTokenCredentialsProvider` function at all? You're not creating a federated token at all anymore

**Comment by mtibben** [comment]
I've taken this further in #1170 - are you able to try that branch?

File: vault/vault.go (hunk 1 above)
Comment (christophgysin1) [reply vault/vault.go:273]: Neither did the previous code. This is simply returning a credentials provider that shall be used to create the federation token.

The previous code went up the chain to the master credentials. This PR changes this to use temporary credentials as defined in the config.

Of course, we can also completely eliminate `NewFederationTokenCredentialsProvider` if you don't think there might be a need for different behaviour for creating federation tokens.

Comment (christophgysin1) [reply vault/vault.go]: Either way, found the right place to change to behaviour for configurations that use `credential_process`. Please see that latest commit.

**Pull request merged**